apis/v1alpha2/httproute_types.go

```diff
@@ -408,6 +408,10 @@ type HTTPQueryParamMatch struct {
 	Value string `json:"value"`
 }
 
+// HTTPMethod describes how to select a HTTP route by matching the HTTP
+// method.
```

Comment:
We should also include a support level here. Given that this is just a shortcut for header matching, I think it probably makes sense to use the same support level we've defined there (Core): https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/httproute_types.go#L328.